dbapi_opentracing/tracing.py

```diff
@@ -40,29 +78,19 @@ def commit(self):
         if not self._self_trace_commit:
             return self.__wrapped__.commit()
 
-        with self._self_tracer.start_active_span(self._self_commit_operation_name) as scope:
-            scope.span.set_tag(tags.DATABASE_TYPE, 'sql')
-            for tag, value in self._self_span_tags.items():
-                scope.span.set_tag(tag, value)
-            return self.__wrapped__.commit()
+        return self._traced_execution(self._self_commit_operation_name, self.__wrapped__.commit)
 
     def rollback(self):
         if not self._self_trace_rollback:
             return self.__wrapped__.rollback()
 
-        with self._self_tracer.start_active_span(self._self_rollback_operation_name) as scope:
-            scope.span.set_tag(tags.DATABASE_TYPE, 'sql')
-            for tag, value in self._self_span_tags.items():
-                scope.span.set_tag(tag, value)
-            return self.__wrapped__.rollback()
-
-    def __enter__(self):
-        return self.cursor()
+        return self._traced_execution(self._self_rollback_operation_name, self.__wrapped__.rollback)
 
     def __exit__(self, exc, value, tb):
         # C extension clients (e.g. psycopg2) require self.__wrapped__.__class__ to be in ConnectionTracing.__bases__,
         # which wrapt doesn't provide.  We need to trace w/ a best-guess operation here instead of passing self to
         # self.__wrapped__.__class__.__exit__() as we can w/ pure python clients.
+        # For psycopg, using PsycopgConnectionTracing is preferred to allow traced commit/rollback in __exit__().
```

Comment:
Is it preferred, or required? (for correctness) If required, can we detect the wrong usage?

Reply:
Required for fully functional connection/cursor.  Updated.